### PR TITLE
WandbLogger's log_image can use step argument

### LINKED ISSUE #11703: Using the step argument raises an error for WandbLogger's log_image
## 🐛 Bug

When using the WandbLogger log_image method, adding the step argument raises an error:
```
  File "/home/nathan/work/these/engine/venv/lib/python3.8/site-packages/pytorch_lightning/loggers/wandb.py", line 417, in log_image
    if len(v) != n:
TypeError: object of type 'int' has no len()
```

### To Reproduce
```

from pytorch_lightning.loggers import WandbLogger
import torch
logger=WandbLogger()
fake_img = torch.randn((50,50))
logger.log_image(key='fake', images=[fake_img], step=0)
```

### Expected behavior

The image is logged with appropriate step in Wandb.

### Environment

- PyTorch Lightning Version (e.g., 1.5.0): stable

### Additional context

Can be easily fixed by swapping a few lines in the method.

cc @awaelchli @morganmcg1 @AyushExel @borisdayma @scottire @manangoel99

### PR DESCRIPTION
**Pull request by NathanGodey**
## What does this PR do?

Makes WandbLogger's log_image compatible with the step argument.

Fixes #11703

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/loggers/wandb.py [hunk 1]
```diff
@@ -413,11 +413,11 @@ def log_image(self, key: str, images: List[Any], **kwargs: str) -> None:
         """
         if not isinstance(images, list):
             raise TypeError(f'Expected a list as "images", found {type(images)}')
+        step = kwargs.pop("step", None)
```
Comment (rohitgr7) [inline pytorch_lightning/loggers/wandb.py:416]: or maybe we could add `step` as a parameter (default None) and keep `kwargs` completely separate?

Comment (rohitgr7) [reply pytorch_lightning/loggers/wandb.py:416]: also mind add an try in the changelog?

Comment (NathanGodey) [reply pytorch_lightning/loggers/wandb.py:416]: I thought it did not introduce any actual change to the expected behaviour, does it make sense to add it anyway ? Like "Make step argument work properly in WandbLogger's log_image" ?

Comment (rohitgr7) [reply pytorch_lightning/loggers/wandb.py]: it's a bug fix, so yes, under the bugfix section.

**Review by carmocca — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

**Comment by awaelchli** [comment]
Thanks for contributing this fix!  <3

**Pull request merged**